packages/core/src/render3/instructions/animation.ts

```diff
@@ -182,15 +191,20 @@ export function ɵɵanimateEnterListener(value: AnimationFunction): typeof ɵɵa
 
   cancelLeavingNodes(tNode, lView);
 
-  getLViewEnterAnimations(lView).push(() => runEnterAnimationFunction(lView, tNode, value));
+  const animations = getLViewEnterAnimations(lView);
+  const enterAnimations = animations.get(tNode.index) ?? [];
+  enterAnimations.push(() => runEnterAnimationFunction(lView, tNode, value));
+  animations.set(tNode.index, enterAnimations);
```

Comment:
nit (as discussed offline):
```suggestion
  registerAnimationFn(getLViewEnterAnimations(lView), () => runEnterAnimationFunction(lView, tNode, value));
```